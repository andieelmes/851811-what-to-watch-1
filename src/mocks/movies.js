export const MOVIES = [
  {
    id: `fantastic-beasts-the-crimes-of-grindelwald`,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
  },
  {
    id: `bohemian-rhapsody`,
    img: `img/bohemian-rhapsody.jpg`,
    title: `Bohemian Rhapsody`,
  },
  {
    id: `seven-years-in-tibet`,
    img: `img/seven-years-in-tibet.jpg`,
    title: `Seven Years in Tibet`,
  },
  {
    id: `what-we-do-in-the-shadows`,
    img: `img/what-we-do-in-the-shadows.jpg`,
    title: `What We Do in the Shadows`,
  },
  {
    id: `pulp-fiction`,
    img: `img/pulp-fiction.jpg`,
    title: `Pulp Fiction`,
  },
];
