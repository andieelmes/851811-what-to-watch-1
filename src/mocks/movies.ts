export const MOVIES = [
  {
    id: 1,
    img: 'https://es31-server.appspot.com/wtw/static/film/background/Midnight_Special.jpg',
    title: 'Midnight Special',
    preview: 'https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4',
    video: 'http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4',
    genre: 'action',
    poster: 'https://es31-server.appspot.com/wtw/static/film/poster/Midnight_Special.jpg',
    backgroundImage: 'https://es31-server.appspot.com/wtw/static/film/background/Midnight_Special.jpg',
    backgroundColor: '#828585',
    description: 'A father and son go on the run, pursued by the government and a cult drawn to the child\'s special powers.',
    rating: 3.3,
    ratingsCount: 67815,
    director: 'Jeff Nichols',
    starring: [
      'Michael Shannon',
      'Joel Edgerton',
      'Kirsten Dunst '
    ],
    duration: 112,
    year: 2016,
    favorite: false
  },
  {
    id: 2,
    img: 'https://es31-server.appspot.com/wtw/static/film/background/What-We-Do-in-the-Shadows.jpg',
    title: 'What We Do in the Shadows',
    preview: 'https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm',
    video: 'http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4',
    genre: 'comedy',
    poster: 'https://es31-server.appspot.com/wtw/static/film/poster/What-We-Do-in-the-Shadows.jpg',
    backgroundImage: 'https://es31-server.appspot.com/wtw/static/film/background/What-We-Do-in-the-Shadows.jpg',
    backgroundColor: '#A39E81',
    description: 'A look into the daily (or rather, nightly) lives of three vampires who\'ve lived together for over 100 years, in Staten Island.',
    rating: 4.2,
    ratingsCount: 6173,
    director: 'Jemaine Clement',
    starring: [
      'Kayvan Novak',
      'Matt Berry',
      'Natasia Demetriou'
    ],
    duration: 30,
    year: 2019,
    favorite: true
  }
];

export const GENRES = [
  `all genres`, `comedies`, `crime`, `documentaries`, `dramas`,
];

export const REVIEWS = [
  {
    id: 1,
    user: {
      id: 13,
      name: 'Zak'
    },
    rating: 2,
    comment: 'A movie that will take you to another world full of emotions.',
    date: '2019-05-30T04:04:34.909Z'
  },
  {
    id: 2,
    user: {
      id: 11,
      name: 'Jack'
    },
    rating: 5,
    comment: 'I really find it difficult to believe this movie is rated highly by people. It\'s hands down the worst movie I\'ve seen in my life',
    date: '2019-06-02T04:04:34.909Z'
  },
  {
    id: 3,
    user: {
      id: 14,
      name: 'Corey'
    },
    rating: 4,
    comment: 'It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie.',
    date: '2019-06-10T04:04:34.909Z'
  }
];
