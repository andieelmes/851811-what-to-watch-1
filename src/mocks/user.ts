export const USER = {
  avatar: '/',
  authorized: true,
  name: 'name'
}
