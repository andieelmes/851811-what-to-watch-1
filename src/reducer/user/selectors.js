import Namespace from "../namespaces";

const NAMESPACE = Namespace.USER;


export const getAuthorizationStatus = (state) => {
  return state[NAMESPACE].isAuthorizationRequired;
};
