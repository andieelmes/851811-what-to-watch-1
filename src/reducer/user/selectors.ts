import Namespace from "../namespaces";

const NAMESPACE = Namespace.USER;


export const getAuthorizationStatus = (state) => {
  return !state[NAMESPACE].isAuthorizationRequired;
};

export const getUserInfo = (state) => {
  const {
    isAuthorizationRequired,
    ...userInfo
  } = state[NAMESPACE];
  return userInfo;
};
