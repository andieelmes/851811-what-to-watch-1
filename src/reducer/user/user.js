import {ENDPOINT_URL} from 'server-variables';

const initialState = {
  isAuthorizationRequired: true,
  avatar: ``,
  name: ``,
};

const ActionType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  GET_USER_DATA: `GET_USER_DATA`
};

const ActionCreator = {
  requireAuthorization: (status) => {
    return {
      type: ActionType.REQUIRED_AUTHORIZATION,
      payload: status,
    };
  },

  getUserData: (data) => {
    return {
      type: ActionType.GET_USER_DATA,
      payload: data,
    };
  },
};

const Operation = {
  login: (email, password) => (dispatch, _getState, api) => {
    return api.post(`/login`, {email, password})
      .then((response) => {
        const data = {
          avatar: `${ENDPOINT_URL}${response.data.avatar_url.replace(`/wtw`, ``)}`,
          name: response.data.name,
        }
        dispatch(ActionCreator.requireAuthorization(false));
        dispatch(ActionCreator.getUserData(data));
      })
      .catch((error) => {
        console.log(error);
      })
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return {...state, isAuthorizationRequired: action.payload}
    case ActionType.GET_USER_DATA: return {...state, ...action.payload}
  }

  return state;
};


export {
  ActionCreator,
  ActionType,
  Operation,
  reducer,
};
