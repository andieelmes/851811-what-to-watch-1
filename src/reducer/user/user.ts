import {ENDPOINT_URL} from 'App/server-variables';
import { Action, ActionCreator, AnyAction } from 'redux';
import { ThunkDispatch, ThunkAction } from 'redux-thunk';

import {User} from 'App/types';
import { Dispatch } from 'react';

const initialState: User = {
  isAuthorizationRequired: true,
  avatar: ``,
  name: ``,
};

const ActionType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  GET_USER_DATA: `GET_USER_DATA`
};

const requireAuthorization: ActionCreator<Action> = (status: boolean) => {
  return {
    type: ActionType.REQUIRED_AUTHORIZATION,
    payload: status,
  };
};

const getUserData: ActionCreator<Action> = (data: {avatar: string, name: string}) => {
  return {
    type: ActionType.GET_USER_DATA,
    payload: data,
  };
};

const ActionCreator = {
  requireAuthorization: requireAuthorization,
  getUserData: getUserData,
};

const onLoginSuccess = (response, dispatch: Dispatch<any>) => {
  const data = {
    avatar: `${ENDPOINT_URL}${response.data.avatar_url.replace(`/wtw`, ``)}`,
    name: response.data.name,
  }
  dispatch(ActionCreator.requireAuthorization(false));
  dispatch(ActionCreator.getUserData(data));
}

const Operation = {
  postLogin: (email: string, password: string, callback: () => void): ThunkAction<void, User, null, AnyAction> => (dispatch: ThunkDispatch<User, void, AnyAction>, _getState: () => User, api: any) => {
    return api.post(`/login`, {email, password})
      .then((response) =>
        {
          onLoginSuccess(response, dispatch);
          callback();
        }
      )
      .catch((error) => {
        console.log(error);
      })
  },
  getLogin: () :ThunkAction<void, User, null, AnyAction> => (dispatch: ThunkDispatch<User, void, AnyAction>, _getState: () => User, api: any) => {
    return api.get(`/login`)
      .then((response) => onLoginSuccess(response, dispatch))
      .catch((error) => {
        console.log(error);
      })
  },
}

const reducer = (state = initialState, action: AnyAction) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return {...state, isAuthorizationRequired: action.payload}
    case ActionType.GET_USER_DATA: return {...state, ...action.payload}
  }

  return state;
};


export {
  ActionCreator,
  ActionType,
  Operation,
  reducer,
};
