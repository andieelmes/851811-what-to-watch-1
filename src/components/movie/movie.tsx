import * as React from 'react';
import { connect } from "react-redux";
import { Link } from 'react-router-dom';
import Profile from 'App/components/profile/profile';
import withActiveItem from 'App/hocs/with-active-item/with-active-item';
import Reviews from 'App/components/reviews/reviews';
import MovieCardList from 'App/components/movie-card-list/movie-card-list';
import Footer from 'App/components/footer/footer';

import { getReviews } from "App/reducer/data/selectors";
import { Operation } from "App/reducer/data/data";

import { Movie as MovieType, Review as ReviewType } from 'App/types';
import { capitalize, getPlural, getRating, getHoursAndMinutes } from 'App/utils';
import { Tabs } from 'App/movie-variables';

interface Props {
  movie: MovieType,
  user: {
    authorized: boolean,
    avatar: string,
    name: string,
  },
  similar: MovieType[],
  onChange: (genre: string) => void
  onLoad: () => ReviewType[],
  activeItem: string,
  reviews: ReviewType[],
};

const MovieCardListWithActiveItem = withActiveItem(MovieCardList);

class Movie extends React.PureComponent<Props> {
  componentDidMount() {
    this.props.onLoad();
  }

  render() {
    const {
      movie,
      user,
      similar,
      onChange,
      activeItem,
      reviews,
    } = this.props;

    const onTabClick = (e, activeItem: string) => {
      e.preventDefault();
      onChange(activeItem);
    };

    return (
      <>
        <section className="movie-card movie-card--full" style={{ backgroundColor: movie.backgroundColor}}>
          <div className="movie-card__hero">
            <div className="movie-card__bg">
              <img src={movie.backgroundImage} alt={movie.title} />
            </div>

            <h1 className="visually-hidden">WTW</h1>

            <header className="page-header movie-card__head">
              <div className="logo">
              <Link to="/" className="logo__link">
                <span className="logo__letter logo__letter--1">W</span>
                <span className="logo__letter logo__letter--2">T</span>
                <span className="logo__letter logo__letter--3">W</span>
              </Link>
              </div>

              <Profile user={user}/>
            </header>

            <div className="movie-card__wrap">
              <div className="movie-card__desc">
                <h2 className="movie-card__title">{movie.title}</h2>
                <p className="movie-card__meta">
                  <span className="movie-card__genre">{capitalize(movie.genre)}</span>
                  <span className="movie-card__year">{movie.year}</span>
                </p>

                <div className="movie-card__buttons">
                  <button className="btn btn--play movie-card__button" type="button">
                    <svg viewBox="0 0 19 19" width="19" height="19">
                      <use xlinkHref="#play-s"></use>
                    </svg>
                    <span>Play</span>
                  </button>
                  <button className="btn btn--list movie-card__button" type="button">
                    <svg viewBox="0 0 19 20" width="19" height="20">
                      <use xlinkHref="#add"></use>
                    </svg>
                    <span>My list</span>
                  </button>
                  <a href="add-review.html" className="btn movie-card__button">Add review</a>
                </div>
              </div>
            </div>
          </div>

          <div className="movie-card__wrap movie-card__translate-top">
            <div className="movie-card__info">
              <div className="movie-card__poster movie-card__poster--big">
                <img src={movie.poster} alt={`${movie.title} poster`} width="218" height="327" />
              </div>

              <div className="movie-card__desc">
                <nav className="movie-nav movie-card__nav">
                  <ul className="movie-nav__list">
                    <li className={`movie-nav__item ${(activeItem === Tabs.OVERVIEW || !activeItem) ? `movie-nav__item--active`: ''}`}>
                      <a href="#" className="movie-nav__link" onClick={(e) => onTabClick(e, Tabs.OVERVIEW)}>Overview</a>
                    </li>
                    <li className={`movie-nav__item ${activeItem === Tabs.DETAILS ? `movie-nav__item--active`: ''}`}>
                      <a href="#" className="movie-nav__link" onClick={(e) => onTabClick(e, Tabs.DETAILS)}>Details</a>
                    </li>
                    <li className={`movie-nav__item ${activeItem === Tabs.REVIEWS ? `movie-nav__item--active`: ''}`}>
                      <a href="#" className="movie-nav__link" onClick={(e) => onTabClick(e, Tabs.REVIEWS)}>Reviews</a>
                    </li>
                  </ul>
                </nav>


                { (activeItem === Tabs.OVERVIEW || !activeItem) && (
                  <>
                    <div className="movie-rating">
                      <div className="movie-rating__score">{movie.rating}</div>
                      <p className="movie-rating__meta">
                        <span className="movie-rating__level">{capitalize(getRating(movie.rating))}</span>
                        <span className="movie-rating__count">{movie.ratingsCount} {getPlural(movie.ratingsCount, 'rating', 'ratings')}</span>
                      </p>
                    </div>

                    <div className="movie-card__text">{movie.description}
                      <p className="movie-card__director"><strong>Director: {movie.director}</strong></p>

                      <p className="movie-card__starring"><strong>Starring: {movie.starring.map((star: string, index: number) => (
                        `${star}${index !== movie.starring.length - 1 && ', '}`
                      ))} and other</strong></p>
                    </div>
                  </>
                )}

                { activeItem === Tabs.DETAILS && (
                  <div className="movie-card__text movie-card__row">
                    <div className="movie-card__text-col">
                      <p className="movie-card__details-item">
                        <strong className="movie-card__details-name">Director</strong>
                        <span className="movie-card__details-value">{movie.director}</span>
                      </p>
                      <p className="movie-card__details-item">
                        <strong className="movie-card__details-name">Starring</strong>
                        <span className="movie-card__details-value">
                          {movie.starring.map((star: string, index: number) => (
                            `${star}${index !== movie.starring.length - 1 && ', '}`
                          ))}
                        </span>
                      </p>
                    </div>

                    <div className="movie-card__text-col">
                      <p className="movie-card__details-item">
                        <strong className="movie-card__details-name">Run Time</strong>
                        <span className="movie-card__details-value">{getHoursAndMinutes(movie.duration)}</span>
                      </p>
                      <p className="movie-card__details-item">
                        <strong className="movie-card__details-name">Genre</strong>
                        <span className="movie-card__details-value">{capitalize(movie.genre)}</span>
                      </p>
                      <p className="movie-card__details-item">
                        <strong className="movie-card__details-name">Released</strong>
                        <span className="movie-card__details-value">{movie.year}</span>
                      </p>
                    </div>
                  </div>
                )}
                { activeItem === Tabs.REVIEWS && <Reviews reviews={reviews}/>}
              </div>
            </div>
          </div>
        </section>

        <div className="page-content">
          { !!similar.length && (
            <section className="catalog catalog--like-this">
              <h2 className="catalog__title">More like this</h2>

              <MovieCardListWithActiveItem movies={similar} withButton/>
            </section>
          )}

          <Footer/>
        </div>
      </>
    );
  };
}

const mapStateToProps = (state, ownProps) => {
  return Object.assign({}, ownProps, {
    reviews: getReviews(state, ownProps.movie.id)
  });
};

const mapDispatchToProps = (dispatch, ownProps) => ({
  onLoad: () => {
    dispatch(Operation.loadComments(ownProps.movie.id));
  },
});

export {Movie};

const connectedMovie = connect(mapStateToProps, mapDispatchToProps)(Movie);

export default connectedMovie;
