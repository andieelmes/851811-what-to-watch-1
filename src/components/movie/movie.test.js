import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import {Movie} from 'App/components/movie/movie';

import {MOVIES} from 'App/mocks/movies';
import {USER} from 'App/mocks/user';

describe(`Movie component`, () => {
  it(`should render correctly`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <Movie
              movie={MOVIES[0]}
              similar={MOVIES}
              onChange={() => {}}
              toggleFavorite={() => {}}
              onPlay={() => {}}
              onLoad={() => {}}
              user={USER}
              activeItem=''
              reviews={[]}
            />
          </BrowserRouter>,
          {createNodeMock: (el) => el}
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
