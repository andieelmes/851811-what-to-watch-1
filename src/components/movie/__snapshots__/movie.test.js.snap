// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie component should render correctly 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": "#828585",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Midnight Special"
          src="https://es31-server.appspot.com/wtw/static/film/background/Midnight_Special.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          className="user-block"
        >
          <a
            href="/mylist"
            onClick={[Function]}
          >
            <div
              className="user-block__avatar"
            >
              <img
                alt="User avatar of name"
                height="63"
                src="/"
                width="63"
              />
            </div>
          </a>
        </div>
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Midnight Special
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Action
            </span>
            <span
              className="movie-card__year"
            >
              2016
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              className="btn movie-card__button"
              href="/film/1/review"
              onClick={[Function]}
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Midnight Special poster"
            height="327"
            src="https://es31-server.appspot.com/wtw/static/film/poster/Midnight_Special.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  href="#"
                  onClick={[Function]}
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              3.3
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Normal
              </span>
              <span
                className="movie-rating__count"
              >
                67815
                 
                ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            A father and son go on the run, pursued by the government and a cult drawn to the child's special powers.
            <p
              className="movie-card__director"
            >
              <strong>
                Director: 
                Jeff Nichols
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring: 
                Michael Shannon, 
                Joel Edgerton, 
                Kirsten Dunst false
                 and other
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          data-movie-id={1}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/film/1"
            onClick={[Function]}
            style={
              Object {
                "display": "block",
              }
            }
          >
            <img
              alt="Midnight Special"
              height="175"
              src="https://es31-server.appspot.com/wtw/static/film/background/Midnight_Special.jpg"
              style={null}
              width="280"
            />
            <video
              controls={true}
              loop={true}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/background/Midnight_Special.jpg"
              style={
                Object {
                  "display": "none",
                }
              }
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/film/1"
              onClick={[Function]}
            >
              Midnight Special
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          data-movie-id={2}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/film/2"
            onClick={[Function]}
            style={
              Object {
                "display": "block",
              }
            }
          >
            <img
              alt="What We Do in the Shadows"
              height="175"
              src="https://es31-server.appspot.com/wtw/static/film/background/What-We-Do-in-the-Shadows.jpg"
              style={null}
              width="280"
            />
            <video
              controls={true}
              loop={true}
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/background/What-We-Do-in-the-Shadows.jpg"
              style={
                Object {
                  "display": "none",
                }
              }
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/film/2"
              onClick={[Function]}
            >
              What We Do in the Shadows
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
